src/dune_console/dune_console.ml

```diff
@@ -189,33 +241,56 @@ module Backend = struct
       let open T in
       let last = ref (Unix.gettimeofday ()) in
       let frame_rate = 1. /. 60. in
+      let cleanup () =
+        state.finished <- true;
+        Base.finish ();
+        Condition.broadcast finish_cv;
+        Mutex.unlock mutex
```

Comment:
We modified this cleanup function to run unconditionally when the event loop terminates.

I imagine that this is technically a bug fix that can prevent some obscure deadlocks with the current UI. Consider the following sequence of events:

1. User requests a shutdown while the rendering loop is sleeping
2. User then immediately closes stderr
3. The rendering loop resumes and tries to render the last pending messages
4. Because the stderr is closed, rendering will fail.
5. As a consequence, the `finish_cv` condition variable will never be broadcasted
6. Dune will be left in a deadlocked stated waiting on this `finish_cv`.

It's probably not worth a separate PR as I don't think this was ever observed in practice.